### PR TITLE
get report's author from request

### PR DESCRIPTION
**Pull request by sylwiawajgel**

### CODE REVIEW

File: server/src/modules/reports/reports.resolver.ts [hunk 1]
```diff
@@ -26,15 +28,16 @@ export class ReportsResolver {
   @Mutation(() => Report)
   @UseGuards(AuthGuard)
   async addReport(
-    @Args('report') reportInput: AddReportInput,
+    @Args('report') reportInput: AddReportInput, @CurrentUser() user: User,
```
Comment (Siemko) [inline server/src/modules/reports/reports.resolver.ts]: `@CurrentUser()` returns `ICurrentUser` interface, not whole `User`.

File: server/src/modules/auth/auth.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { AuthService } from './auth.service';
+
+@Resolver()
+export class AuthResolver {
+  constructor(
+    @Inject(AuthService) private readonly authService: AuthService,
+  ) {}
+
+  @Query(() => TokenModel)
+  async refreshToken(@Args('token') token: string): Promise<TokenModel> {
```
Comment (MichalLytek) [inline server/src/modules/auth/auth.resolver.ts]: `refreshToken` should be a mutation?

Comment (sylwiawajgel) [reply server/src/modules/auth/auth.resolver.ts]: So what is the definition of a mutation? I thought it is an interference in database and generating jwt token is not.

Comment (MichalLytek) [reply server/src/modules/auth/auth.resolver.ts]: Mutations is an equivalent of POST/PUT/DELETE from REST. Have you implemented issuing token in REST API as `GET /token` or maybe using `POST`? 😉 
In terms of query and mutation distinction, you need to think about caching, e.g. when you fetch list of posts, it doesn't have to be super fresh as new posts may appear a few times in a day or a week, so you can omit the request and serve the data from apollo cache. Issuing token using a query operation means that you might receive the same token as in previous call (served from cache), so it won't be refreshed 😄 
Also you might have to modify the system state by e.g. counting failed attempts to be able to block user after N times, etc.

Comment (sylwiawajgel) [reply server/src/modules/auth/auth.resolver.ts]: Ok, thank you for explanation! Now everything is clear 😀

**Pull request closed without merging**